pubsub/publisher.go

```diff
@@ -0,0 +1,81 @@
+package pubsub
+
+import (
+	"cloud.google.com/go/pubsub"
+	"context"
+	"encoding/json"
+	"errors"
+	queuesgo "github.com/merlinapp/queues-go"
+	"reflect"
+)
+
+type publisher struct {
+	topic      *pubsub.Topic
+	objectType reflect.Type
+}
+
+/*
+Creates a new Google's pubsub publisher
+the topic must exists already on the given project
+the objectType interface should be any of the following types, any other type will cause an error returning a nil value
+1. Copy of a structure
+2. No nil pointer of a structure
+3. A map with key string and any value
```

Comment:
also, some minor typos :) 

```
The topic must already exist in the given project.
...
2. Non-nil pointer to a struct of the expected type.
...
```

And a question, when do we want to publish a `map` instead of a `struct` ? we can talk about it offline, I recall you mentioned it to me a few hours ago, but it didn't click at that point haha.

Reply:
Done! thanks